Source/UI/NewUI/PaymentForm/View/CardNumber/CardNumberViewModel.swift

```diff
@@ -34,18 +36,15 @@ extension CardNumberViewModel: CardNumberViewModelProtocol {
     let cardNumber = cardUtils.removeNonDigits(from: rawText)
 
     switch cardValidator.validateCompleteness(cardNumber: cardNumber) {
-    case .failure:
-        return nil
-    // An unknown scheme can also be generated when an eager validation was matched
-    //    but the number is not complete.
-    // In this case we only update delegate if we have a final result to avoiding
-    //    overriding the eager validation with a less informative update
-    case .success((let isComplete, let scheme)):
-      guard isComplete else {
-          return nil
-      }
-      delegate?.update(cardNumber: cardNumber, scheme: scheme)
-      return Constants.Bundle.SchemeIcon(scheme: scheme)
+        // An unknown scheme can also be generated when an eager validation was matched
+        //    but the number is not complete.
+        // In this case we only update delegate if we have a final result to avoiding
+        //    overriding the eager validation with a less informative update
+      case .success((let isComplete, let scheme)):
+        guard isComplete, supportedSchemes.contains(scheme) else { return nil }
+        delegate?.update(cardNumber: cardNumber, scheme: scheme)
+        return Constants.Bundle.SchemeIcon(scheme: scheme)
+      case .failure: return nil
```

Comment:
```suggestion
      case let .success((isComplete, scheme)) where isComplete && supportedSchemes.contains(scheme):
        delegate?.update(cardNumber: cardNumber, scheme: scheme)
        return Constants.Bundle.SchemeIcon(scheme: scheme)
      default: return nil
```

Reply:
My main point is that we can integrate the check for `isComplete` and `supportedSchemes.contains(scheme)` into the switch, cleaning up the branching here. Could do something like this instead:
```suggestion
      case let .success((isComplete, scheme)) where isComplete && supportedSchemes.contains(scheme):
        delegate?.update(cardNumber: cardNumber, scheme: scheme)
        return Constants.Bundle.SchemeIcon(scheme: scheme)
      case .success, .failure: return nil
```